mreg_cli/api/models.py

```diff
@@ -55,13 +55,77 @@
 from mreg_cli.utilities.shared import convert_wildcard_to_regex
 from mreg_cli.utilities.validators import is_valid_category_tag, is_valid_location_tag
 
-_mac_regex = re.compile(r"^([0-9A-Fa-f]{2}[.:-]){5}([0-9A-Fa-f]{2})$")
+IPNetMode = Literal["ipv4", "ipv6", "ip", "network", "networkv4", "networkv6"]
 
 
 class NetworkOrIP(BaseModel):
     """A model for either a network or an IP address."""
 
-    ip_or_network: str | IP_AddressT | IP_NetworkT
+    ip_or_network: IP_AddressT | IP_NetworkT
+
+    @classmethod
+    def validate(cls, value: str | IP_AddressT | IP_NetworkT | Self) -> Self:
```

Comment:
As this returns an instance, is this `validate()` or `new()`? It feels a little weird to call it for instantiation.

Reply:
If this was Go, I would definitely have named it `New()`, however, there isn't an agreed-upon naming scheme for alternate constructors in Python as far as I'm aware, so that passed me by. I decided to use the now-deprecated Pydantic v1 method name `validate()` which takes in any value, validates it, and returns an instance of the model.

We can change the name if you want. I don't mind either way. I just needed something better than `from_string`, since we can validate a lot more than just strings.

Reply:
Let's call it a nod to Pydantic v1 and keep it. :)